### PR TITLE
ROX-27073: script for multi cluster test setup

### PR DESCRIPTION
**Pull request by johannes94**
## Description

This PR:
- Changes available script for dev/CI environment setup to allow deploying only data plane components by setting `DATAPLANE_ONLY="true"`
- Adds a script that deploys the fleet* stack to 2 pre-existing clusters, one with control plane components the other with only data plane componets

This is a prerequisite for the setup for multi cluster migration testing. The actual test execution and automated creation of infra cluster will follow in additional PRs.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [x] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [x] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources
- [ ] (If applicable) Changes to the dp-terraform Helm values have been reflected in the addon on integration environment

## Test manual

**TODO:** Add manual testing efforts

```
# Setup 2 infractl flavor OSD on AWS
# After the clusters are ready get the kubeconfig and store them in your local filesystem
mkdir ~/kubes
url=$(infractl artifacts jm-migration-1 --json | jq '.Artifacts[] | select(.Name=="kubeconfig") | .URL' -r)
wget -O ~/kubes/cluster1 $url

url=$(infractl artifacts jm-migration-2 --json | jq '.Artifacts[] | select(.Name=="kubeconfig") | .URL' -r)
wget -O ~/kubes/cluster2 $url

# Setup env vars required for execution
export CLUSTER_1_KUBECONFIG="~/kubes/cluster1"
export CLUSTER_2_KUBECONFIG="~/kubes/cluster2"
export CLUSTER_TYPE="infra-openshift"

bash scripts/ci/multicluster_tests/run_multicluster_tests.sh
# Local execution will prompt you for kinit and quay credentials
# Verify:
# 1. the scripts executes successfully
# 2. Cluster 1 has fleet-manager and fleetshard-sync running
# 3. Cluster 2 has only fleetshard-sync running
# 4. Cluster 2's fleetshard-sync is connected to cluster 1's fleet-manager

```

### CODE REVIEW

File: dev/env/scripts/up.sh [hunk 1]
```diff
@@ -58,12 +52,21 @@ if ! is_openshift_cluster "$CLUSTER_TYPE"; then
     make -C "$GITROOT" deploy/redhat-pull-secret
 fi
 
-log "Deploying fleet-manager"
-make -C "$GITROOT" deploy/service
+DATAPLANE_ONLY=${DATAPLANE_ONLY:-}
+if [[ "${DATAPLANE_ONLY}" == "" ]]; then
```
Comment (kovayur) [inline dev/env/scripts/up.sh:56]:
```suggestion
if [[ -z "${DATAPLANE_ONLY}" ]]; then
```

File: dev/env/scripts/up.sh [hunk 2]
```diff
@@ -90,25 +93,28 @@ fi
 
 # Sanity check.
 wait_for_container_to_become_ready "$ACSCS_NAMESPACE" "app=fleetshard-sync" "fleetshard-sync" 500
-# Prerequisite for port-forwarding are pods in ready state.
-wait_for_container_to_become_ready "$ACSCS_NAMESPACE" "app=fleet-manager" "service"
 
-if [[ "$ENABLE_EMAIL_SENDER" == "true" ]]; then
-    wait_for_container_to_become_ready "$ACSCS_NAMESPACE" "application=emailsender" "emailsender"
-fi
+if [[ "${DATAPLANE_ONLY}" == "" ]]; then
```
Comment (kovayur) [inline dev/env/scripts/up.sh:97]:
```suggestion
if [[ -z "${DATAPLANE_ONLY}" ]]; then
```

File: scripts/ci/multicluster_tests/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+# as well need additional / other commands
+make deploy/bootstrap
+make deploy/dev
+
+# service template for dev defines a reencrypt route which requires manual creation of a self
+# signed certificate before starting fleet-manager. We don't want that which is why we're seeting
+# termination to edge for this route
+kubectl patch -n rhacs route fleet-manager -p '{"spec":{"tls":{"termination":"edge"}}}'
+FM_URL="https://$(kubectl get routes -n rhacs fleet-manager -o yaml | yq .spec.host)"
+export FM_URL
```
Comment (kovayur) [inline scripts/ci/multicluster_tests/deploy.sh:29]: Do you really need export here and for `CLUSTER_1_KUBECONFIG`, `CLUSTER_2_KUBECONFIG`?

File: scripts/ci/multicluster_tests/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+# Restart fleet-manager to pickup the config
+kubectl delete pod -n rhacs -l app=fleet-manager
+
+export KUBECONFIG="$CLUSTER_2_KUBECONFIG"
+make deploy/bootstrap
+DATAPLANE_ONLY="true" make deploy/dev
+
+# Get a static token from cluster1 which will be used for FS -> FM communication
+# for the FS running on cluster2
+STATIC_TOKEN=$(KUBECONFIG="$CLUSTER_1_KUBECONFIG" kubectl create token -n rhacs fleetshard-sync --audience acs-fleet-manager-private-api --duration 8760h)
```
Comment (kovayur) [inline scripts/ci/multicluster_tests/deploy.sh:52]: Can't fleet manager trust both clusters at the same time?

**Review by kovayur — APPROVED** [review]
Nice, just a few minor comments, feel free to ignore.

File: dev/env/scripts/up.sh (hunk 1 above)
Comment (johannes94) [reply dev/env/scripts/up.sh:56]: Done.

File: dev/env/scripts/up.sh (hunk 2 above)
Comment (johannes94) [reply dev/env/scripts/up.sh:97]: Done.

File: scripts/ci/multicluster_tests/deploy.sh (hunk 3 above)
Comment (johannes94) [reply scripts/ci/multicluster_tests/deploy.sh:29]: Yes, I need it to be available in some child process in this one and also in later on once I add the test execution.

File: scripts/ci/multicluster_tests/deploy.sh (hunk 4 above)
Comment (johannes94) [reply scripts/ci/multicluster_tests/deploy.sh:52]: Yes I think we could do that as well. I felt like adding trust for both clusters at the same time is more complicated then just using this static token.

**Pull request merged**